src/web/wms/src/main/java/org/geoserver/wms/web/publish/WMSLayerConfig.java

```diff
@@ -239,5 +247,47 @@ private void initWMSCascadedUI(IModel<LayerInfo> layerModel) {
 
         remoteFormatsPalette.add(new DefaultTheme());
         remoteForamtsContainer.add(remoteFormatsPalette);
+        // scale denominators
+        TextField<Integer> minScale =
+                new TextField(
+                        "minScale",
+                        new PropertyModel<Boolean>(wmsLayerInfo, "minScale"),
+                        Integer.class);
+        metaDataCheckBoxContainer.add(minScale);
+        TextField<Integer> maxScale =
+                new TextField(
+                        "maxScale",
+                        new PropertyModel<Boolean>(wmsLayerInfo, "maxScale"),
+                        Integer.class);
+        metaDataCheckBoxContainer.add(maxScale);
+
+        minScale.add(new ScalesValidator(minScale, maxScale));
```

Comment:
Useful. I'd also add a check that they are both non negative (there should be a built-in validator for that, for ranges of numbers if memory serves me right).

Reply:
sure will check for negatives also